### PR TITLE
Some additional options for supervision-related methods

### PR DESCRIPTION
**Pull request by desh2608**
This PR adds some options that may be useful in manipulating supervisions/cuts, such as:
* CLI method for `SupervisionSet.with_alignment_from_ctm()`
* `max_segment_duration` option in `CutSet.trim_to_alignments()` to control the length of segments
* `merge_policy` option in `merge_supervisions()` --> this allows either concatenating with a delimiter (default), or using the first supervision's value.

I will add test cases for these.

### CODE REVIEW

File: lhotse/cut/mono.py [hunk 1]
```diff
@@ -248,16 +251,22 @@ def merge_supervisions(
                 f"recognition models (cut id: {self.id})."
             )
 
+        merge_func_ = partial(
```
Comment (pzelasko) [inline lhotse/cut/mono.py:254]: remember to handle `merge_custom` as well

File: lhotse/cut/base.py [hunk 2]
```diff
@@ -510,14 +510,16 @@ def trim_to_alignments(
...
         delimiter: str = " ",
         keep_all_channels: bool = False,
     ) -> "CutSet":  # noqa: F821
         """
         Splits the current :class:`.Cut` into its constituent alignment items (:class:`.AlignmentItem`).
         These cuts have identical start times and durations as the alignment item. Additionally,
         the `max_pause` option can be used to merge alignment items that are separated by a pause
-        shorter than `max_pause`.
+        shorter than `max_pause`. If `max_segment_duration` is specified, the merging will be
+        performed only if the resulting segment is shorter than `max_segment_duration`.
```
Comment (pzelasko) [inline lhotse/cut/base.py:522]: This doc suggests to me that when the resulting segment is longer than `max_segment_duration`, the items would be left unmerged; that's not true, is it?

**Review by pzelasko — COMMENTED** [review]
Cool! I left a couple of comments.

File: lhotse/cut/base.py [hunk 3]
```diff
@@ -700,6 +727,7 @@ def cut_into_windows(
         duration: Seconds,
         hop: Optional[Seconds] = None,
         keep_excessive_supervisions: bool = True,
+        use_alignment_if_exists: Optional[str] = None,
```
Comment (desh2608) [inline lhotse/cut/base.py:730]:
```suggestion
```

File: lhotse/cut/base.py [hunk 4]
```diff
@@ -712,10 +740,25 @@ def cut_into_windows(
         :param hop: Shift between the windows in the new cuts in seconds.
         :param keep_excessive_supervisions: bool. When a cut is truncated in the
             middle of a supervision segment, should the supervision be kept.
+        :param use_alignment_if_exists: Optional str. If provided, the corresponding alignments will
+            be used to cut the supervisions according to the time. This could mean that resulting
+            cut durations are slightly different than the requested ``duration``, since we will
+            try to align the supervisions to the alignment boundaries.
```
Comment (desh2608) [inline lhotse/cut/base.py:746]:
```suggestion
```

File: lhotse/cut/base.py [hunk 5]
```diff
@@ -712,10 +740,25 @@ def cut_into_windows(
...
+            # Only check the first supervision (checking all would be too slow)
+            assert (
+                use_alignment_if_exists in self.supervisions[0].alignment
+            ), f"Supervision does not have alignment of type {use_alignment_if_exists}"
+            return self.cut_into_windows_with_alignment(
+                alignment_type=use_alignment_if_exists,
+                duration=duration,
+                hop=hop,
+            )
+
```
Comment (desh2608) [inline lhotse/cut/base.py:761]:
```suggestion
```

**Review by pzelasko — APPROVED** [review]
LGTM!

**Pull request merged**